client/src/main/java/org/apache/crail/rpc/RpcDispatcher.java

```diff
@@ -134,6 +135,12 @@ public RpcDispatcher(ConcurrentLinkedQueue<RpcConnection> connectionList) {
 		return connections[0].pingNameNode();
 	}
 
+	@Override
+	public RpcFuture<RpcRemoveDataNode> removeDataNode(
+			InetAddress ipaddr, int port) throws Exception {
+		return connections[0].removeDataNode(ipaddr, port);
```

Comment:
This will not work in a multi-namenode environment. We would have to send a removeDataNode to all Namenodes. For now we should at least check if there is only one namenode and throw an exception otherwise (e.g. datanode removal only possible with one namenode at the moment).